File: 프로그래머스/3/49189. 가장 먼 노드/가장 먼 노드.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] edge) {
        
        ArrayList<ArrayList<Integer>> node = new ArrayList<>();
        
        for(int i=0; i<=n; i++){
            node.add(new ArrayList<>());
        }
        
        for(int i=0; i<edge.length; i++){
            int from = edge[i][0];
            int to = edge[i][1];
            node.get(from).add(to);
            node.get(to).add(from);
        }

        boolean visited[] = new boolean[n+1];
        visited[1] = true;
        
        Deque<Integer> que = new ArrayDeque<>();
        que.add(1);
        
        int size = 0;
        
        while(!que.isEmpty()){
            size = que.size();
            for(int i=0; i<size; i++){
                int now = que.poll();
                for(int next : node.get(now)){
                    if(visited[next]) continue;
                    visited[next] = true;
                    que.add(next);
                }
            }
        }
        
        return size;
    }
}

File: 프로그래머스/3/49189. 가장 먼 노드/가장 먼 노드.js
function solution(n, edge) {
    
    const node = Array.from({length : n+1}, () => []);
    
    for(let i=0; i<edge.length; i++){
        let from = edge[i][0];
        let to = edge[i][1];
        node[from].push(to);
        node[to].push(from);
    }
    
    const visited = Array(n+1).fill(false);
    visited[1] = true;
    
    const que = [1];
    var size;
    
    while(que.length){
        var size = que.length;
        for(let i=0; i<size; i++){
            var now = que.shift();
            for(let next of node[now]){
                if(visited[next]) continue;
                visited[next] = true;
                que.push(next);
            }
        }
    }
    
    return size;
}